File: tools/tal_support/as274_fc/F/t4.f
      program test4
c
c     Test of subroutine INV.
c
      double precision rbar(21), rinv(15), rho, zero
      integer pos, row, col
c
      data rho/0.6/, zero/0.0/
c
      pos = 1
      do 20 row = 1, 6
	do 10 col = row+1, 7
	  if (col .eq. row+1) then
	    rbar(pos) = rho
	  else
	    rbar(pos) = zero
	  end if
	  pos = pos + 1
   10   continue
   20 continue
      call inv(7, 21, rbar, 6, rinv)
C I added the "21" argument.  It's supposed to be nrbar.
C     - ajay shah
      write(*, 900) rinv
  900 format(1x, 5g14.6/ 15x, 4g14.6/ 29x, 3g14.6/ 43x, 2g14.6/
     +      57x, g14.6)
      end

File: tools/tal_support/as274_fc/C/t4.c
#include <stdio.h>
#include "as274c.h"
#include "misc.h"

int main()
{
    double *rbar, *rinv, rho=.6;
    int pos=0, row, col;

    rbar = dvector(0, 20);
    rinv = dvector(0, 14);

    for (row=0; row<6; row++)
      for (col=row+1; col<7; col++)
        rbar[pos++] = col == row+1 ? rho : 0;

    inv(7, 21, rbar, 6, rinv);
    pr_utdm_v(rinv, 5, 13, 7);
    return 0;
}
